doc/source/performance/jemalloc-profiling.rst

```diff
@@ -0,0 +1,230 @@
+
+.. list-table::
+    :widths: 20 40
+    :header-rows: 1
+
+    * - Option
+      - Description
+    * - `--enable-stats`
+      - Enables statistics-gathering ability
+    * - `--enable-prof`
+      - Enables heap profiling and the ability to detect leaks. 
+
+Using ``LD_PRELOAD``. Build the library, configure the malloc configuration with the ``prof:true`` string, and then use ``LD_PRELOAD`` to  preload the ``libjemalloc.so`` library. The libprocess ``MemoryProfiler`` class detects the library automatically and enables the profiling support.
+
+The following is an example of the required commands:
+
+.. sourcecode:: bash
+
+    ./configure --enable-stats --enable-prof && make && make install
+    MALLOC_CONF=prof:true 
+    LD_PRELOAD=/usr/lib/libjemalloc.so
+
+.. note:
+
+    Ensure the ``libjemalloc.so`` exists in the LD path.
+
+Usage
+-------
+To detect if jemalloc is set, run the following command:
+
+.. sourcecode::  bash
+
+    jemalloc_detected ON
+
+To enable jemalloc profiling in a MySQL client, run the following command:
+
+.. sourcecode:: bash
+
+    set global jemalloc_profiling=on;
+
+
+The :ref:`malloc_stats_totals` table returns the statistics, in bytes, of the memory usage. The command takes no parameters and returns the results as a table.
+
+The following example commands display this result:
+
+.. sourcecode:: mysql
+
+    use performance_schema;
+
+    SELECT * FROM malloc_stats_totals;
+    +----+------------+------------+------------+-------------+------------+
+    | id | ALLOCATION | MAPPED     | RESIDENT   | RETAINED    | METADATA   |
+    +----+------------+------------+------------+-------------+------------+
+    |  1 | 390977528  | 405291008  | 520167424  | 436813824   | 9933744    |
+    +----+------------+------------+------------+-------------+------------+
+    1 row in set (0.00 sec)
+
+The :ref:`malloc_stats` table returns the cumulative totals, in bytes, of allocation requests and allocation returns. The command takes no parameters and returns the results as a table.
+
+The following example commands display this result:
+
+.. sourcecode:: mysql
+
+    use performance_schema;
+
+    mysql> SELECT * FROM malloc_stats ORDER BY TYPE DESC LIMIT 3;
+    +--------+-------------+-------------+-------------+-------------+
+    | TYPE   | ALLOCATED   | NMALLOC     | NDALLOC     | NRESQUESTS  |
+    +--------+-------------+-------------+-------------+-------------+
+    | small  | 23578872    | 586156      | 0           | 2649417     |
+    | large  | 367382528   | 2218        | 0           | 6355        |
+    | huge   | 0           | 0           | 0           | |
+    +--------+-------------+-------------+-------------+-------------+
+    3 rows in set (0.00 sec)
+
+The profiling samples the ``malloc()`` calls and stores the sampled stack traces in a separate location in memory. These samples can be dumped into the filesystem. A dump returns a detailed view of the state of the memory. 
```

Comment:
Maybe I would create a sub section for this i.e. `Dumping the memory profile`

Reply:
Added